pkg/controller.v1beta1/experiment/experiment_controller_status.go

```diff
@@ -20,12 +20,13 @@ import (
 	"context"
 
 	experimentsv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/experiments/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type updateStatusFunc func(instance *experimentsv1beta1.Experiment) error
 
 func (r *ReconcileExperiment) updateStatus(instance *experimentsv1beta1.Experiment) error {
-	err := r.Status().Update(context.TODO(), instance)
+	err := r.Status().Update(context.TODO(), instance, &client.SubResourceUpdateOptions{UpdateOptions: client.UpdateOptions{FieldManager: "experiment-controller"}})
```

Comment:
@kubeflow/wg-training-leads @gjyotin305 Can we also use the server side apply for other resources that we update in our controllers.
For instance:
Suggestion CR: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/suggestion_controller_status.go#L36
Trial CR: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/trial/trial_controller_status.go#L45

@tenzen-y Do we need to update k8s version for server-side apply for Katib ?